prusti-viper/src/encoder/mir/procedures/encoder/mod.rs

```diff
@@ -1781,13 +1781,46 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         encoded_statements: &mut Vec<vir_high::Statement>,
     ) -> SpannedEncodingResult<()> {
         let block = &self.mir[bb];
-        if self.try_encode_specification_function_call(bb, block, encoded_statements)? {
+        if self.try_encode_assert(bb, block, encoded_statements)? {
+            Ok(())
+        } else if self.try_encode_specification_function_call(bb, block, encoded_statements)? {
             Ok(())
         } else {
             unreachable!()
         }
     }
 
+    fn try_encode_assert(
+        &mut self,
+        bb: mir::BasicBlock,
+        block: &mir::BasicBlockData<'tcx>,
+        encoded_statements: &mut Vec<vir_high::Statement>,
+    ) -> SpannedEncodingResult<bool> {
+        for stmt in &block.statements {
+            if let mir::StatementKind::Assign(box (
+                _,
+                mir::Rvalue::Aggregate(box mir::AggregateKind::Closure(cl_def_id, cl_substs), _),
+            )) = stmt.kind
```

Comment:
Can you tell these apart from loop invariants? Maybe if the `get_prusti_assertion_specs` call returned `None`, this should return `Ok(false)`?